### DISCUSSION TITLE
How to regenerate scaffolding?

### DISCUSSION THREAD

**Discussion by rumstead** [kubernetes-sigs/kubebuilder#2864] [Support] [answered]
Let's say I want to pick up the newest scaffolding after upgrading kubebuilder, is there a command I can run to regenerate the scaffolding?

I tried:
1. Deleting a file generated by a template (MakefIle)
    * it did not regenerate
1. `kubebuilder create` with the same params

**Comment by camilamacedo86** [comment, marked as answer]
All commands executed with the tool (data) are tracked in the PROJECT file.
(ihmo) it would be desirable to have a command that re-scaffolds all based on that so it could help in the migration process ( like I want to upgrade my project to the latest versions changes ) 

We would have an alpha command to try to make it where we could inform via a flag the PROJECT file and the --plugins which we would like to use to do the re-scaffold. Then it would mean for example:

$ kubebuilder alpha scaffold --project-config=<inform the path of the PROJECT file> --plugins=go/v4,grafana/alphav1 --output=path

That would result in:
-> create a directory with the project name
-> call the tool to init the project with the plugins --plugins=go/v4,grafana/alphav1
-> call edit subCommans for edit the project ( example if multigroup be enabled )
-> call create API to create all APIs and controllers scaffold (we also need to check if the api was not scaffold with for example an specific plugin like deploy-image/alphav1)
-> call create webhook to create the webhooks scaffold as well

That also can lead to future changes/additions to the PROJECT file in case we discover that we are missing info to be tracked there. 

Would you like to work on this one?
Would you like to contribute to this one?

See https://github.com/kubernetes-sigs/kubebuilder/issues/2068 . 

I am adding this idea suggestion there as well.

**Reply by rumstead** [reply]
I am very interested in contributing it

**Reply by camilamacedo86** [reply]
Hi @rumstead,

For sure, we would like to make easier the process to upgrade the projects to ensure that they have the latest changes.
In the past, it was what make me advocate in favor we track all in the PROJECT file. (ihmo) An alpha command with some implementation like that could let us try to see if it can work, what the users think about it if it is achievable and etc.

If you be able to do a POC, that would be great if you could demo it in the community meeting and get the other's input. Btw, your attendance in the community is very welcome either. On top of that, if you have any other ideas/suggestions to address this requirement please feel free to raise and share them.

**Reply by rumstead** [reply]
will do. does this look to be on the correct track? https://github.com/kubernetes-sigs/kubebuilder/compare/master...rumstead:kubebuilder:master?expand=1

**Reply by camilamacedo86** [reply]
> will do. does this look to be on the correct track? https://github.com/kubernetes-sigs/kubebuilder/compare/master...rumstead:kubebuilder:master?expand=1

Yes, you can create PR with all against the master.
Then, you can demo that in the community meeting.
For you create an alpha sub-command that would be here: https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/cli/alpha

**Comment by camilamacedo86** [comment]
Closing since it is answered.
